stage0/src/zero_page.rs

```diff
@@ -132,7 +132,7 @@ impl ZeroPage {
         // Construct the "real" E820 table, carving out a chunk of memory out for the ACPI area and
         // reserved memory just below 1 MiB.
         for entry in &e820_table[0..e820_entries] {
-            if entry.addr() < 0x8_0000 && entry.addr() + entry.size() >= 0x10_0000 {
+            if entry.addr() <= 0x8_0000 && entry.addr() + entry.size() >= 0x10_0000 {
```

Comment:
Was there an issue here? If yes, then we probably have some other conditions below that need changing as well.

Reply:
I just noticed the if statement in line 137 will always evalute to true if there wasn't a `<=` here. I plan on revisiting all of this as part of the next PR, so will double-check all the logic.